File: WorkoutTracker/src/app/shared/models/exercise-set.ts
import { ExerciseType } from "src/app/exercises/models/exercise-type";
import { DbModel } from "./db-model";
import { TimeSpan } from "./time-span";
import { ModelWeight } from "src/app/shared/models/model-weight";

export interface ExerciseSet extends DbModel {
    totalWeight: ModelWeight|null;
    totalTime: TimeSpan|null;
    totalReps: number|null;

    weight: ModelWeight|null;
    time: TimeSpan|null;
    reps: number|null;

    exerciseId: number;
    exerciseName: string;
    exerciseType: ExerciseType;
    exercisePhoto: string|null;
}

File: WorkoutTracker.Domain/Entities/Exercises/ExerciseSets/ExerciseSet.cs
using WorkoutTracker.Domain.Base;
using WorkoutTracker.Domain.ValueObjects;

namespace WorkoutTracker.Domain.Entities.Exercises.ExerciseSets;

public class ExerciseSet : IDbModel
{
    public long Id { get; set; }

    public ModelWeight? Weight { get; set; }
    public TimeSpan? Time { get; set; }
    public int? Reps { get; set; }

    public long ExerciseId { get; set; }
    public Exercise? Exercise { get; set; }

    public long ExerciseSetGroupId { get; set; }
    public ExerciseSetGroup? ExerciseSetGroup { get; set; }
}
